pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -59,11 +59,11 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) u
 		}
 		errs = append(errs, validateCommonQueueSort(profilesPath, cc.Profiles)...)
 	}
-	for _, msg := range validation.IsValidSocketAddr(cc.HealthzBindAddress) {
-		errs = append(errs, field.Invalid(field.NewPath("healthzBindAddress"), cc.HealthzBindAddress, msg))
+	if len(cc.HealthzBindAddress) > 0 {
```

Comment:
I think we should just make users of v1beta1 API pass this validation. To do that, we can empty the fields during conversion. And we can remove the fields from the internal type.

That way, the fields truly become no-op.

Reply:
I think we need to validate here if we are to validate that the port flag is zero.

Reply:
agree on making this validation match the --port validation (can be unset or can be an address with an explicitly "0" port)

Reply:
this comment is still unresolved

Reply:
@liggitt updated, I added the check for the address validity (as it was before) and check for the port to be 0 if set